### ISSUE TITLE
Fail to open pck with correct encryption key

### ISSUE THREAD

**Issue by Snoogey** [bruvzg/gdsdecomp#143]
### System information

Windows 10, Godot 4.1

### Issue description

Unable to open Godot 4.1 pck file with correct encryption key.
Encrypted key fetched directly from project exe that's loading said pck.

I can share the pck + encryption key if that helps.

### Recovery log

-2023-09-02-20-24-55 [Read PCK]: Error opening PCK file: C:/encrypted.pck
Opening file: C:/encrypted.pck
USER ERROR: FATAL ERROR: Failed to load project pack!
   at: GDRESettings::load_pack (modules\gdsdecomp\utility\gdre_settings.cpp:380)
USER ERROR: Transient parent has another exclusive child.
   at: Window::set_visible (scene\main\window.cpp:747)

**Comment by nikitalita** [comment]
> I can share the pck + encryption key if that helps.
Yeah, I’m going to need to test this myself. You can share it either here or shoot me a discord DM. I’m “nikita_lita” in the official Godot discord: https://discord.gg/4JBkykG

**Comment by nikitalita** [comment]
It turns out this was a chain-loaded pck that used the config file encryption scheme. We don't currently support this. I am reticent to add a feature to decrypt files like this because it could be used to decrypt save files; encrypted save files are usually used for Steam achievements, which would mean messing with them would mess with external systems. I do not want to encourage that; it would create potential headaches for devs and would be an escalation in the obfuscation arms race.

**Comment by KoxaKoxama** [comment]
@nikitalita Can you please share some info how is this config file encryption scheme is working? Not related to this specific pck, I would like to try this chain-loaded pck method of encryption for myself.
Thanks!

**Comment by awerc** [comment]
@nikitalita I have the same issue with Backpack Battles game from Steam, but with another error `ERROR: The MD5 sum of the decrypted file does not match the expected value. It could be that the file is corrupt, or that the provided decryption key is invalid.` Thank you in advance if you can check what this problem is related to.

**Comment by nikitalita** [comment]
I took a look, Backpack Battles uses a custom encryption scheme that we don't support; I can't tell what they're doing. I would suggest coordinating with @awerc if people are interested in figuring out how to decrypt the scripts for that game.

**Comment by MirkoCovizzi** [comment]
> I took a look, Backpack Battles uses a custom encryption scheme that we don't support; I can't tell what they're doing. I would suggest coordinating with [@awerc](https://github.com/awerc) if people are interested in figuring out how to decrypt the scripts for that game.

You don't need to RE the custom encryption scheme. Just attach a debugger and step through until right before the call to the decryption function (you can see which one from the engine source code since it uses the same one, the custom key manipulation code before doesn't matter). Then you just need to read the data in the memory address used in the function call. I will not go into more detail into how to do it, and this comment is for educational purposes only.

**Comment by curealb** [comment]
> [@nikitalita](https://github.com/nikitalita) I have the same issue with Backpack Battles game from Steam, but with another error `ERROR: The MD5 sum of the decrypted file does not match the expected value. It could be that the file is corrupt, or that the provided decryption key is invalid.` Thank you in advance if you can check what this problem is related to.

Hello, did you solve this problem? It seems not used the official encryption. I used IDA to find 'GDScript::load_byte_code' and got the key '8671424952921006D39F4C9E5E8F821391E2B06A80D946D693FB8757154CE849', but it doesn’t seem to be correct.
